storage/objects/list_files_all_version.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objects
+
+// [START storage_list_file_archived_generations]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+// listFilesAllVersion lists both live and noncurrent versions of objects within specified bucket.
+func listFilesAllVersion(w io.Writer, bucket string) error {
```

Comment:
I think we should avoid the term "files" for GCS objects if possible (I know other samples already use this, but...). The corresponding Java sample is called `ListObjectsWithOldVersions`, let's use something similar here?